treescript/mercurial.py

```diff
@@ -0,0 +1,196 @@
+"""Treescript mercurial functions."""
+# import asyncio
+# from asyncio.subprocess import PIPE, STDOUT
+# import functools
+# import hashlib
+# import json
+import logging
+import os
+import sys
+# from shutil import copyfile
+# import traceback
+# from collections import namedtuple
+
+from treescript.utils import execute_subprocess
+from treescript.exceptions import FailedSubprocess
+from treescript.task import get_source_repo, get_tag_info
+
+# https://www.mercurial-scm.org/repo/hg/file/tip/tests/run-tests.py#l1040
+# For environment vars.
+
+HGRCPATH = os.path.join(os.path.dirname(__file__), 'data', 'hgrc')
+ROBUSTCHECKOUT_PATH = os.path.abspath(
+    os.path.join(os.path.dirname(__file__), 'py2', 'robustcheckout.py')
+)
```

Comment:
These two are candidates for config file fodder, so we can update them via puppet.